src/joint_state_listener.cpp

```diff
@@ -56,23 +57,104 @@ JointStateListener::JointStateListener(const KDL::Tree& tree, const MimicMap& m)
   // set publish frequency
   double publish_freq;
   n_tilde.param("publish_frequency", publish_freq, 50.0);
+
   // get the tf_prefix parameter from the closest namespace
   std::string tf_prefix_key;
   n_tilde.searchParam("tf_prefix", tf_prefix_key);
   n_tilde.param(tf_prefix_key, tf_prefix_, std::string(""));
-  publish_interval_ = ros::Duration(1.0/max(publish_freq,1.0));
+  publish_interval_ = ros::Duration(1.0 / max(publish_freq, 1.0));
+
+
+  /// only offer reload_service on request
+  bool with_reload;
+  n_tilde.param("with_reload_service", with_reload, false);
+  if (with_reload)
+    {
+      reload_server = n_tilde.advertiseService("reload_robot_model", &JointStateListener::reload_robot_model_cb, this);
+    }
+
 
   // subscribe to joint state
   joint_state_sub_ = n.subscribe("joint_states", 1, &JointStateListener::callbackJointState, this);
 
   // trigger to publish fixed joints
-  timer_ = n_tilde.createTimer(publish_interval_, &JointStateListener::callbackFixedJoint, this);
+  pub_fixed_trafos_timer_ = n_tilde.createTimer(publish_interval_, &JointStateListener::callbackFixedJoint, this);
+  pub_fixed_trafos_timer_.start();
+}
 
-};
 
 
-JointStateListener::~JointStateListener()
-{};
+bool loadTreeAndMimicMap(KDL::Tree  tree, MimicMap *mimic_map, std::string *err_msg)
```

Comment:
Double spaces between `KDL::Tree` and `tree` looks strange. Should this be a reference instead?